include/zephyr/zbus/zbus.h

```diff
@@ -881,13 +889,34 @@ struct zbus_observer_node {
  *                or one of the special values K_NO_WAIT and K_FOREVER.
  *
  * @retval 0 Observer added to the channel.
+ * @retval -EEXIST The observer is already present in the channel's observers list.
  * @retval -EALREADY The observer is already present in the channel's runtime observers list.
  * @retval -EAGAIN Waiting period timed out.
  * @retval -EINVAL Some parameter is invalid.
+ * @retval -EBUSY The node is already in use.
  */
-int zbus_chan_add_obs(const struct zbus_channel *chan, const struct zbus_observer *obs,
-		      struct zbus_observer_node *node, k_timeout_t timeout);
+int zbus_chan_add_obs_with_node(const struct zbus_channel *chan, const struct zbus_observer *obs,
+				struct zbus_observer_node *node, k_timeout_t timeout);
```

Comment:
It should be made clear in the description that this function only works if no allocator is selected. You can turn this or the other function into a `static inline` in the header and return `-ENOTSUP` to make it more clear.